### ISSUE TITLE
OMV 5 - Error occurs changing Web Administration Settings

### ISSUE THREAD

**Issue by HannesJo0139** [openmediavault/openmediavault#472]
Opening a new issue here since I am not able to reopen #463. Guess I have no permission.

In #463 you said the message "An error has occurred" when activating SSL is normal. I disagree. I am connected via HTTP and I have not set "force SSL". So HTTP connection should still be ok.

I am using port 70 for HTTP and 400 for HTTPS. If I'm connected via HTTPS and turn it off or change its port, connection gets lost - That is totally clear. But if I am connected via HTTP and change HTTPS port to lets say 401 I also get "An error has occurred". That should not happen, since I have not changed anything regarding to HTTP settings.

In fact I can not do any change in that panel without getting this error message.

**Comment by votdev** [comment]
This is a normal behavior because if you press apply, the config files are build and the nginx process is restarted. If there is a request during this period, then it will fail.

**Comment by HannesJo0139** [comment]
Maybe better to change that behavior then

**Comment by votdev** [comment]
How?

**Comment by votdev** [comment]
IMO any effort to workaround this normal behavior is not worth because as mentioned it is no bug that a request fails when the webserver is restarted and second you do this configuration step only once.

**Comment by HannesJo0139** [comment]
For example show the user a notice and hide that special error. Imagine using a coffe machine and every time u use the function 'coffee' it displays "an error has occured" just because lets say the milk module was not used but expected to. Usability = 0. 
If it is expected to dont get a connection, the gui should handle that and not show an error where the behavior is no error but normal.

**Comment by votdev** [comment]
It's not possible to find out from the UI when the nginx webserver is restarted, so the UI can't ignore the error in such a case. Submitting the changes of the webserver settings and applying them (by pressingthe apply button) are totally independent.

But as told, i'm open for any contribution.

**Comment by HannesJo0139** [comment]
I appreciate that! Ok how about just putting a notice on that settings page, like 
„Note: The web server must be restarted. You may receive an error message because it cannot respond during this time.“

In the meantime, I can familiarize myself with the OMV code and if I can think of a really good solution, I will send a pull request.

**Comment by HannesJo0139** [comment]
Yes, I don't think anyone likes such solutions, but it would be an acceptable emergency solution as long as there is no real one. I understand now how the behaviour is created, but many people do not. At least not right away. Just putting the hint in the documentation might be a good separation of the software components, but as a usability advocate I really think you should make an exception here.

Back to my example with the coffee machine. Let's say the machine always shows an error if you just brew a simple black coffee because the milk module is not used but the software thinks it should have been used. Every user will be surprised about this false error. Maybe the customer will try it again and again because he thinks there was an error. Now the manufacturer comes and says "Yes, we know the behavior, have a look at the manual. It says so." That won't change the fact that many customers are standing there and think something has gone wrong.

**Comment by shalak** [comment]
I'd like to put my 3 cents, I hope you don't mind: if I knew about this known issue, I wouldn't have lost time to look for errors in logs.

> I do not like such solutions. I think a better place for this is the documentation.

Sure, not a clean solution, but I believe it's better than plain "An error has occurred" with no explanation. Why do you think that it's better to keep such crucial information in documentation? Googling for the produced error message doesn't point to this issue, nor to the documentation, unless someone explicitly adds "applying web administration settings" (or similar) to the query. 

How about adding an error ID to the message, so it's easier to find this known fact of OMV?

**Comment by votdev** [comment]
> Sure, not a clean solution, but I believe it's better than plain "An error has occurred" with no explanation. Why do you think that it's better to keep such crucial information in documentation? Googling for the produced error message doesn't point to this issue, nor to the documentation, unless someone explicitly adds "applying web administration settings" (or similar) to the query.

It's not possible to add an explanation because the backend and frontend does not know that the web server is down at the given time.

> How about adding an error ID to the message, so it's easier to find this known fact of OMV?

Does not make sense because the error appears out of the scope of OMV, so it is not possible to set an error code.

All i can do is to add this information to the [documentation](https://openmediavault.readthedocs.io/en/latest/administration/general/workbench.html).

**Comment by shalak** [comment]
> Does not make sense because the error appears out of the scope of OMV, so it is not possible to set an error code.

How about tuning the error message into "A web interface error has occurred" for any web interface errors? Such error, when changing web interface settings, would be obvious to be expected...
